Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -269,6 +270,7 @@ public final class Chat {
     ///
     /// - Throws: An error while sending a message to the Stream API.
     public func resendMessage(_ messageId: MessageId) async throws {
+        let messageEditor = try client.backgroundWorker(of: MessageEditor.self)
```

Comment:
After going back and forth I ended up with this solution which is accessing background workers only when needed and throwing an error if it is missing (See `ChatClient.createBackgroundWorkers`).